godot-core/src/obj/traits.rs

```diff
@@ -44,6 +44,13 @@ where
         Self::class_id()
     }
 
+    /// Class icon resource path. Will use this icon if available.
+    ///
+    /// You can also set an icon via `#[class(icon = "path/to/icon.svg")]`.
+    fn icon() -> &'static str {
+        ""
+    }
```

Comment:
Not 100% sure if this should be in the `GodotClass` trait... there is quite a few registration properties that aren't currently exposed like this (e.g. `#[class(internal)]`). For builder APIs, it could otherwise be a separate step:
```rs
builder.class::<MyClass>()
    .icon("path/to/icon.svg")
    .internal()
    .build()
```

---

But _if_ we decide `GodotClass` is the right place -- which needs further discussion -- then:

1. This should document how the path has to look, i.e.
   - relative to where
   - whether it should start with `res://` or not
   - maybe an example

2. Could be named `editor_icon`. Since trait methods can be called unqualified, and user-defined classes having their own `icon` isn't that unlikely.